### PR TITLE
Enforce cargo fmt in azure pipelines

### LINKED ISSUE #68: Deny warnings and enforce cargo fmt in azure pipelines
We should enforce `cargo fmt` and deny warnings in CI! Plenty of projects do this. [Wasmer does this with circle CI](https://github.com/wasmerio/wasmer/blob/master/Makefile#L30), and I'm sure we can figure it out for azure pipelines.

### PR DESCRIPTION
**Pull request by andrew**
Fixes #68

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ jobs:
           containerResource: rust133
     container: $[ variables['containerResource'] ]
     steps:
+      - script: |
+          export PATH=$PATH:$HOME/.cargo/bin
+          cargo fmt --all -- --check
```
Comment (xmclark) [inline azure-pipelines.yml]:
```suggestion
          rustup component add rustfmt
          cargo fmt --all -- --check
```

**Comment by andrew** [comment]
Tests are passing now 🎉

**Comment by xmclark** [comment]
Sorry to do this, but can you also add the check to the very recent windows job? It got merged last night. Thank you contribution! 🙏

**Comment by andrew** [comment]
Thanks for the fix!

**Pull request merged**